app/services/spam_rule/auto_silence.rb

```diff
@@ -1,37 +1,38 @@
 class SpamRule::AutoSilence
 
-  def initialize(user)
-    @user = user
-  end
+  attr_reader :group_message
 
-  def self.silence?(user)
-    self.new(user).silence?
+  def initialize(user, post = nil)
+    @user = user
+    @post = post
   end
 
-  def self.punish!(user)
-    self.new(user).silence_user
+  def perform
+    I18n.with_locale(SiteSetting.default_locale) do
+      silence_user if should_autosilence?
+    end
   end
 
-  def perform
-    silence_user if silence?
+  def self.prevent_posting?(user)
+    user.blank? || user.silenced? || new(user).should_autosilence?
   end
 
-  def silence?
-    return true if @user.silenced?
+  def should_autosilence?
+    return false if @user.blank?
     return false if @user.staged?
     return false if @user.has_trust_level?(TrustLevel[1])
 
-    if SiteSetting.num_spam_flags_to_silence_new_user > (0) &&
-        SiteSetting.num_users_to_silence_new_user > (0) &&
-        num_spam_flags_against_user >= (SiteSetting.num_spam_flags_to_silence_new_user) &&
-        num_users_who_flagged_spam_against_user >= (SiteSetting.num_users_to_silence_new_user)
+    if SiteSetting.num_spam_flags_to_silence_new_user > 0 &&
```

Comment:
This is a minor style change. I don't know why you'd want to wrap the right side of a compare in parens.

Reply:
Why didn't Rubocop pick that up?

Reply:
Not sure, I'll take a look today and see if that's an option.